### PR TITLE
feat: Add Astra DB Tool components for use with Tool Agents

### PR DESCRIPTION
**Pull request by smatiolids**
This PR introduces two types of tools for Langflow, enabling data from AstraDB/Cassandra tables and collections to be integrated with Tool Calling Agents.

The following files were added:

- AstraCQLTool.py: The Astra CQL tool configures a tool that can query data from a table using the AstraDB REST API.
- AstraTool.py: The Astra tool configures a tool to query data from collections using the AstraDB DATA API.

A subtle change in base/agents/agent.py was applied to allow inputting tools created as "StructuredTool."

Documentation for the components is added to components-tools.md.

### CODE REVIEW

**Comment by ogabrielluiz** [comment]
These look awesome, @smatiolids 

Thank you!

**Review by ogabrielluiz — APPROVED** [review]
LGTM

File: src/backend/base/langflow/components/tools/AstraCQLTool.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import requests
+from langchain.pydantic_v1 import BaseModel, Field, create_model
+from langchain_core.tools import StructuredTool
+
+from langflow.base.langchain_utilities.model import LCToolComponent
+from langflow.io import DictInput, IntInput, SecretStrInput, StrInput
+from langflow.schema import Data
+
+
+class AstraCQLToolComponent(LCToolComponent):
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/tools/AstraCQLTool.py:13]: Thoughts on `AstraDBCQLToolComponent`? (Adding `DB` into the names of the tools)

File: src/backend/base/langflow/components/tools/AstraCQLTool.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            is_list=True,
+            info="Field name and description to the model",
+        ),
+        DictInput(
+            name="static_filters",
+            display_name="Static Filters",
+            is_list=True,
+            info="Field name and value. When filled, it will not be generated by the LLM.",
+        ),
+        IntInput(name="limit", display_name="Limit", value=5),
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/tools/AstraCQLTool.py:80]: For better or worse, we've gone with a more verbose naming in astra db component. Do you mind double-checking the similar parameters have the same naming / documentation? https://github.com/langflow-ai/langflow/blob/main/src/backend/base/langflow/components/vectorstores/AstraDB.py#L178

File: src/backend/base/langflow/components/tools/AstraCQLTool.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    ]
+
+    def astra_rest(self, args):
+        headers = {"Accept": "application/json", "X-Cassandra-Token": f"{self.token}"}
+        ASTRA_URL = f"{self.api_endpoint}/api/rest/v2/keyspaces/{self.keyspace}/{self.table}/"
+        key = []
+        # Partition keys are mandatory
+        for k in self.partition_keys.keys():
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/components/tools/AstraCQLTool.py:88]: There is a make target, `make format` (and `make lint`) you can run to see if they catch these - we've made some ruff changes recently

File: src/backend/base/langflow/components/tools/AstraCQLTool.py (hunk 1 above)
Comment (smatiolids) [reply src/backend/base/langflow/components/tools/AstraCQLTool.py:13]: Agreed. I will rename files and classes.

File: src/backend/base/langflow/components/tools/AstraCQLTool.py (hunk 2 above)
Comment (smatiolids) [reply src/backend/base/langflow/components/tools/AstraCQLTool.py:80]: Ok. I adjusted some names, like limit.

File: src/backend/base/langflow/components/tools/AstraCQLTool.py (hunk 3 above)
Comment (smatiolids) [reply src/backend/base/langflow/components/tools/AstraCQLTool.py:88]: Ok. I ran format and lint locally.

**Pull request merged**